src/checkout/context/checkout.tsx

```diff
@@ -119,27 +122,35 @@ export const CheckoutProvider: FC<Props> = React.memo(({children}) => {
   const getBillingSettings = useCallback(async () => {
     try {
       const resp = await getSettingsNotifications({})
+      if (resp.status === 404) {
+        // We're injesting the 404 error here and leaving the default inputs since
+        // Quartz's API returns a 404 for valid requests that don't have any data
+        // Meaning, if a user hasn't filled out the notification settings, the
+        // API response is expected to return a 404 even though the query was successful
+        return
+      }
       if (resp.status !== 200) {
         throw new Error(resp.data.message)
       }
-      setInputs({
-        ...inputs,
+      setInputs(prevInputs => ({
+        ...prevInputs,
         balanceThreshold: resp.data.balanceThreshold,
         notifyEmail: resp.data.notifyEmail,
         shouldNotify: resp.data.isNotify,
-      })
+      }))
     } catch (error) {
-      // We're injesting the error here and leaving the default inputs since
-      // Quartz's API returns a 404 for valid requests that don't have any data
-      // Meaning, if a user hasn't filled out the notification settings, the
-      // API response is expected to return a 404 even though the query was successful
-      console.error(error)
+      const message = getErrorMessage(error)
+      dispatch(notify(getBillingSettingsError(message)))
     }
-  }, [inputs])
+  }, [dispatch])
 
   useEffect(() => {
     getBillingSettings()
-  }, [getBillingSettings])
+    return () => {
+      // Call the `quartz/me` endpoint to update the existing user metadata
+      dispatch(getQuartzMeThunk())
```

Comment:
we're going to want to refresh the user's status only after they leave the page (if they've upgraded) since we want for them to stay on the page after the upgrade to see the success overlay and go from there